### PR TITLE
refactor: expose generated code and client business logic to share with Kapacitor

### PR DESCRIPTION
**Pull request by lesam**

### CODE REVIEW

**Comment by lesam** [comment]
Depends on: https://github.com/influxdata/influx-cli/pull/102

**Comment by danxmoran** [comment]
Holding off on a full review until #102 merges and this rebases, to make it easier to see the truly new stuff.

Can you summarize all the moves? I see:
* `internal/api` -> `api`
* `internal/cmd` -> `clients`

Any others?

File: clients/bucket/delete_test.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package bucket_test
 import (
 	"bytes"
 	"context"
+	"github.com/influxdata/influx-cli/v2/clients"
```
Comment (danxmoran) [inline clients/bucket/delete_test.go:6]: Bad import sort (I'd expect `make fmt` to fix it, let me know if not)

File: internal/mock/gen.go [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+//go:generate go run github.com/golang/mock/mockgen -package mock -destination api_organizations.gen.go github.com/influxdata/influx-cli/v2/api OrganizationsApi
+//go:generate go run github.com/golang/mock/mockgen -package mock -destination api_setup.gen.go github.com/influxdata/influx-cli/v2/api SetupApi
+//go:generate go run github.com/golang/mock/mockgen -package mock -destination api_write.gen.go github.com/influxdata/influx-cli/v2/api WriteApi
+//go:generate go run github.com/golang/mock/mockgen -package mock -destination api_query.gen.go github.com/influxdata/influx-cli/v2/api QueryApi
+//go:generate go run github.com/golang/mock/mockgen -package mock -destination api_users.gen.go github.com/influxdata/influx-cli/v2/api UsersApi
+//go:generate go run github.com/golang/mock/mockgen -package mock -destination api_delete.gen.go github.com/influxdata/influx-cli/v2/api DeleteApi
 
 // Other mocks
 //go:generate go run github.com/golang/mock/mockgen -package mock -destination config.gen.go -mock_names Service=MockConfigService github.com/influxdata/influx-cli/v2/internal/config Service
```
Comment (danxmoran) [inline internal/mock/gen.go:15]: Should the config service also move out of `internal`?

Comment (lesam) [reply internal/mock/gen.go:15]: I don't need it for now, I vote we leave it in `internal`

File: clients/bucket/delete_test.go (hunk 1 above)
Comment (lesam) [reply clients/bucket/delete_test.go:6]: Nope, `make fmt` was already run on this.

**Review by danxmoran — APPROVED** [review]
